### DISCUSSION TITLE
How can I add ports to all elements in an array created by grid function?

### DISCUSSION THREAD

**Discussion by GiaQuyetNgo** [gdsfactory/gdsfactory#3069] [Q&A]
Dear GDSfactory experts,
I have created an array in a diagonal direction and the first element already has 4 ports namely: 'o1', 'o2', 'o3', and 'o4'. However, when I make an array using gf.grid(), I could not see the ports from any element. Even the ports of the first element are not shown.
What can I do to enable or add ports for all elements in the array? Do the ports automatically have different names?
I want to route all input and output ports as lanes to the outermost box, can you suggest a good practice for this purpose?
Thank you very much!
![1](https://github.com/user-attachments/assets/7a971fba-6ca9-43dd-a612-f14c050fe0e7)

**Comment by joamatab** [comment]
im not sure i follow,

```python
import gdsfactory as gf 

c = gf.grid([gf.components.straight, gf.components.bend_euler])
c.pprint_ports()
```

gives you
```
┏━━━━━━━━┳━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━┓
┃ name   ┃ width ┃ orientation ┃ layer ┃ center           ┃ port_type ┃
┡━━━━━━━━╇━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━┩
│ 0_0_o1 │ 0.5   │ 180.0       │ WG    │ (3.25, 8.25)     │ optical   │
│ 0_0_o2 │ 0.5   │ 0.0         │ WG    │ (13.25, 8.25)    │ optical   │
│ 1_0_o1 │ 0.5   │ 180.0       │ WG    │ (18.375, 3.375)  │ optical   │
│ 1_0_o2 │ 0.5   │ 90.0        │ WG    │ (28.375, 13.375) │ optical   │
└────────┴───────┴─────────────┴───────┴──────────────────┴───────────┘
```
